### PR TITLE
- adds azure pipelines spfx cicd documentation

### PR DESCRIPTION
**Pull request by baywet**
#### Category
- [ ] Content fix
- [x] New article

#### Related issues:
none

#### What's in this Pull Request?
- adds new documentation to leverage Azure Pipelines with SPFx for CI, CT, CD

### CODE REVIEW

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+ms.prod: sharepoint
+localization_priority: Priority
+---
+# Implement Continuous Integration and Continuous deployment using Azure Pipelines
+Azure Pipelines is the newer version of the Azure DevOps build and release features.  
+
+This article explains the steps involved in setting up your Azure Pipelines environment with Continuous Integration and Continuous Deployment to automate your SharePoint Framework builds, unit tests, and deployment.  
+
+> [!NOTE] 
+> Azure Multi-stage Pipelines is still in preview and you need to make sure you enable this preview feature for your Azure DevOps organization under _Preview Features_
```
Comment (andrewconnell) [inline docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: Can you please make "Preview Features" bold? Menu items should be bold.

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+## Chosing between Azure Multi-stage Pipelines (preview) and Azure DevOps builds and releases
+There are currently two approaches available to implement Continuous Integration, Testing and Deployement in Azure DevOps.  
+Azure builds and releases is the historic one, featuring a graphical edition experience and storing the definitions in a json document hidden from the user.
```
Comment (andrewconnell) [inline docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: json should be JSON

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+# Implement Continuous Integration and Continuous deployment using Azure Pipelines
+Azure Pipelines is the newer version of the Azure DevOps build and release features.  
+
+This article explains the steps involved in setting up your Azure Pipelines environment with Continuous Integration and Continuous Deployment to automate your SharePoint Framework builds, unit tests, and deployment.  
+
+> [!NOTE] 
+> Azure Multi-stage Pipelines is still in preview and you need to make sure you enable this preview feature for your Azure DevOps organization under _Preview Features_
+
+## Chosing between Azure Multi-stage Pipelines (preview) and Azure DevOps builds and releases
+There are currently two approaches available to implement Continuous Integration, Testing and Deployement in Azure DevOps.
```
Comment (andrewconnell) [inline docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: CI/CD should be lower case, not caps. lcase is how you have it in other places in this PR

Also... isn't "testing" part of integration? Never seen it sep. out as it's own thing... let's not introduce it as something new here.

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+> [!NOTE] 
+> You can comment out/remove the _PublishCodeCoverageResults_, _PublishTestResults_ and _npm test_ tasks if you do not have any unit test implement and/or do not want unit tests to run.
+
+> [!NOTE] 
+> You can find the latest version of this file on the [sample](https://github.com/SharePoint/sp-dev-build-extensions/tree/master/samples/azure-devops-ci-cd-spfx)
+
+## Implementing Continuous Deployment
+The continuous deployment stage is described by the following YAML template.  
+Copy the following content in a new file at the root of the project called `azure-pipelines-deploy-template.yml`.
```
Comment (andrewconnell) [inline docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: Files should be BOLD, not inline code. Inline code is reserved for code.

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+> You can find the latest version of this file on the [sample](https://github.com/SharePoint/sp-dev-build-extensions/tree/master/samples/azure-devops-ci-cd-spfx)
+> [!NOTE] 
+> You can define multiple environments and by duplicating the `deployqa` stage and providing different parameters. If you do so, make sure the stage name, the job name, the target environment and the variable group name are unique.
+> [!NOTE] 
+> You can conditionnally deploy to different environments leveraging [conditions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml)
+
+## Configuring the credentials for the environments
+Secrets should never be committed to a repository for security reasons. The pipeline described in the previous steps makes use of _variable groups_ to keep configuration values secret. The variable groups need to be created for each environment and the name needs to match what is described in the pipeline definition (here `qa_configuration`).  
+To create the variable group, follow these steps:
+1. Logon to Azure DevOps, navigate to your project
```
Comment (andrewconnell) [inline docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: Please change "login" => "sign-in"

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+1. Under _Pipelines_ click on _Library_
```
Comment (andrewconnell) [inline docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: Please change:
- menu items to BOLD, not italic. 
- click => select (accessibility reasons)

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md (hunk 1 above)
Comment (baywet) [reply docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: fixed

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md (hunk 2 above)
Comment (baywet) [reply docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: fixed

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md (hunk 3 above)
Comment (baywet) [reply docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: fixed

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md (hunk 4 above)
Comment (baywet) [reply docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: fixed

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md (hunk 5 above)
Comment (baywet) [reply docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: fixed

File: docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md (hunk 6 above)
Comment (baywet) [reply docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md]: fixed

**Comment by baywet** [comment]
@andrewconnell thanks for the review, implemented the changes, let me know if there's anything else.

**Pull request merged**